common/src/wdt.rs

```diff
@@ -0,0 +1,83 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    wdt.rs
+
+Abstract:
+
+    File contains execution routines for programming the Watchdog Timer.
+
+Environment:
+
+    ROM
+
+--*/
+
+use caliptra_drivers::SocIfc;
+use core::cmp::max;
+
+use crate::cprintln;
+
+pub struct WdtTimeout(pub u32);
+
+impl Default for WdtTimeout {
+    fn default() -> WdtTimeout {
+        WdtTimeout(WDT1_TIMEOUT_SECS)
+    }
+}
+
+const EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES: u64 = 20_000_000; // 20 million cycles
+const WDT1_TIMEOUT_SECS: u32 = 5;
+const WDT2_TIMEOUT_CYCLES: u64 = 1; // Fire immediately after WDT1 expiry
+
+/// Start the Watchdog Timer
+///
+/// # Arguments
+///
+/// * `soc_ifc` - SOC Interface
+///
+///
+pub fn start_wdt(soc_ifc: &mut SocIfc, timeout: WdtTimeout) {
+    cprintln!("[state] Starting the Watchdog Timer");
+
+    let wdt1_timeout_cycles: u64 = match soc_ifc.get_cycle_count(timeout.0) {
+        Ok(cycle_count) => max(EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES * 10, cycle_count),
+        Err(_) => EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES * 10,
+    };
+
+    // Set WDT1 period.
+    soc_ifc.set_wdt1_timeout(wdt1_timeout_cycles);
+
+    // Set WDT2 period.
+    soc_ifc.set_wdt2_timeout(WDT2_TIMEOUT_CYCLES);
+
+    // Enable WDT1 only. WDT2 is automatically scheduled (since it is disabled) on WDT1 expiry.
+    soc_ifc.configure_wdt1(true);
+}
+
+/// Restart the Watchdog Timer
+///
+/// # Arguments
+///
+/// * `env` - ROM Environment
+///
+pub fn restart_wdt(soc_ifc: &mut SocIfc) {
+    cprintln!("[state] Restarting the Watchdog Timer");
```

Comment:
I wonder if we should remove these printlns. For a long-running system, I worry these will pollute the UART logs.

Reply:
Good point.